WordPress/WordPressShareExtension/ShareModularViewController.swift

```diff
@@ -826,35 +857,21 @@ fileprivate extension ShareModularViewController {
                 self.tracks.trackExtensionError(error)
                 self.showAlert()
             })
-        } else {
-            // No media. just a simple post
-            networkService.saveAndUploadPost(title: shareData.title,
-                                             body: shareData.contentBody,
-                                             tags: shareData.tags,
-                                             categories: shareData.selectedCategoriesIDString,
-                                             status: shareData.postStatus.rawValue,
-                                             siteID: siteID,
-                                             onComplete: {
-                                                self.tracks.trackExtensionPosted(self.shareData.postStatus.rawValue)
-                                                self.dismiss()
-            }, onFailure: {
-                let error = self.createErrorWithDescription("Failed to save and upload post with no media.")
-                self.tracks.trackExtensionError(error)
-                self.showAlert()
-            })
         }
     }
 
-    func showAlert() {
-        let title = NSLocalizedString("Sharing Error", comment: "Share extension error dialog title.")
-        let message = NSLocalizedString("Whoops, something went wrong while sharing. You can try again, maybe it was a glitch.", comment: "Share extension error dialog text.")
+    func showAlert(title: String = NSLocalizedString("Sharing Error", comment: "Share extension error dialog title."),
+                   message: String = NSLocalizedString("Whoops, something went wrong while sharing. You can try again, maybe it was a glitch.", comment: "Share extension error dialog text."),
+                   retry: Bool = true) {
```

Comment:
Instead of passing a retry `Bool`, maybe we could simply pass in a completion handler that will get executed after the alert is dismissed. This will avoid baking-in any specific logic within this function.